File: src/minecraft/me/skrilled/utils/friend/FriendManager.java
package me.skrilled.utils.friend;

import me.skrilled.SenseHeader;
import me.skrilled.utils.config.impl.ConfigFriends;

import java.util.LinkedList;

public class FriendManager {

    private final LinkedList<Friend> friends = new LinkedList<>();

    public FriendManager() {
        this.getFriends().clear();
    }

    public void addFriend(Friend friend) {
        this.getFriends().add(friend);
        SenseHeader.getSense.getConfigManager().save(new ConfigFriends());
    }

    public void removeFriend(Friend friend) {
        this.getFriends().remove(friend);
        SenseHeader.getSense.getConfigManager().save(new ConfigFriends());
    }

    public Friend getFriendByName(String name) {
        return this.getFriends().stream().filter(friend -> friend.getName().equalsIgnoreCase(name)).findFirst().orElse(null);
    }

    public LinkedList<Friend> getFriends() {
        return friends;
    }
}

File: minecraft/me/kl0udy92/apart/features/friends/FriendManager.kt
package me.kl0udy92.apart.features.friends

import me.kl0udy92.apart.config.implementations.FriendsConfig
import me.kl0udy92.apart.core.Main

class FriendManager {

    val friends = mutableListOf<Friend>()

    init {
        this.friends.clear()
    }

    fun addFriend(friend: Friend) {
        this.friends.add(friend)
        Main.configManager.write(FriendsConfig())
    }

    fun removeFriend(string: String) {
        this.friends.remove(this.getFriendByName(string))
        Main.configManager.write(FriendsConfig())
    }

    fun getFriendByName(name: String): Friend? {
        return this.friends.stream().filter { it.name.equals(name, true) }.findFirst().orElse(null)
    }

}